include/irr/asset/CTexturePacker.h

```diff
@@ -4,35 +4,96 @@
 #include "irr/asset/format/EFormat.h"
 #include "irr/asset/ICPUImage.h"
 #include "irr/core/alloc/GeneralpurposeAddressAllocator.h"
+#include "irr/core/math/morton.h"
 
 namespace irr {
 namespace asset
 {
 
+//TODO move this to different file
+template<typename addr_type>
+class CSquareAddressAllocator
```

Comment:
you need to inherit from generalpurpose, inheritance not composition.

And you definitely don't want ot be using linear allocator (we want to use this and support adding and removing squares at runtime).

Reply:
still relevant